pkg/providers/vsphere/vmprovider_resourcepolicy_test.go

```diff
@@ -178,6 +178,22 @@ func resourcePolicyTests() {
 					Expect(resourcePolicy.Status.ClusterModules).To(HaveExactElements(status.ClusterModules))
 					assertSetResourcePolicy(resourcePolicy, true)
 				})
+
+				It("should create additional cluster modules", func() {
+					assertSetResourcePolicy(resourcePolicy, true)
+					resourcePolicy.Spec.ClusterModuleGroups = append(resourcePolicy.Spec.ClusterModuleGroups, "another-NodeGroup2")
+					Expect(vmProvider.CreateOrUpdateVirtualMachineSetResourcePolicy(ctx, resourcePolicy)).To(Succeed())
+					Expect(resourcePolicy.Status.ClusterModules).To(HaveLen(len(resourcePolicy.Spec.ClusterModuleGroups) * len(ctx.ZoneNames)))
```

Comment:
Doesn't assertSetResourcePolicy() already assert this (it also takes into account for when ClustersPerZone > 1)?

Reply:
Yeah, you are right. I was thinking about this to make sure we see the creation, but this already happens in that assert function.